File: Scripts/results/generated_code_results/baseline_gpt/Finance/Barclays/HelpScreen.js
import React from 'react';
import { View, Text, TextInput, StyleSheet, TouchableOpacity, Image } from 'react-native';
import Icon from 'react-native-vector-icons/MaterialIcons';

const HelpScreen = () => {
  return (
    <View style={styles.container}>
      {/* Header */}
      <Text style={styles.headerText}>Help</Text>

      {/* Search Bar */}
      <View style={styles.searchContainer}>
        <Icon name="search" size={24} color="#aaa" />
        <TextInput 
          style={styles.searchInput}
          placeholder="Search or ask a question"
          placeholderTextColor="#aaa"
        />
      </View>

      {/* Quick Links */}
      <Text style={styles.sectionTitle}>Quick links</Text>
      <View style={styles.quickLinksContainer}>
        <QuickLink icon="error" text="Report fraud" />
        <QuickLink icon="credit-card" text="Cards" />
        <QuickLink icon="article" text="Statements" />
      </View>

      {/* Help Topics */}
      <Text style={styles.sectionTitle}>Help topics</Text>
      <HelpTopic title="Browse all topics" description="Check how to manage Direct Debits, pay in a cheque and more." />

      {/* More Help */}
      <Text style={styles.sectionTitle}>More help</Text>
      <HelpTopic title="Money worries and support" />
      <HelpTopic title="Accessible services" />
      <HelpTopic title="Our service status" />
      <HelpTopic title="Tour the app" />
    </View>
  );
};

const QuickLink = ({ icon, text }) => (
  <TouchableOpacity style={styles.quickLink}>
    <Icon name={icon} size={40} color="#007bff" />
    <Text style={styles.quickLinkText}>{text}</Text>
  </TouchableOpacity>
);

const HelpTopic = ({ title, description }) => (
  <TouchableOpacity style={styles.helpTopic}>
    <View>
      <Text style={styles.helpTopicTitle}>{title}</Text>
      {description && <Text style={styles.helpTopicDescription}>{description}</Text>}
    </View>
    <Icon name="chevron-right" size={24} color="#aaa" />
  </TouchableOpacity>
);

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    paddingHorizontal: 16,
    paddingTop: 24,
  },
  headerText: {
    fontSize: 32,
    fontWeight: 'bold',
    marginBottom: 16,
  },
  searchContainer: {
    flexDirection: 'row',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
    borderRadius: 8,
    padding: 10,
    marginBottom: 16,
  },
  searchInput: {
    marginLeft: 10,
    flex: 1,
  },
  sectionTitle: {
    fontSize: 18,
    fontWeight: 'bold',
    marginVertical: 16,
  },
  quickLinksContainer: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    marginBottom: 24,
  },
  quickLink: {
    alignItems: 'center',
  },
  quickLinkText: {
    marginTop: 8,
    textAlign: 'center',
  },
  helpTopic: {
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'center',
    backgroundColor: '#f5f5f5',
    borderRadius: 8,
    padding: 16,
    marginBottom: 8,
  },
  helpTopicTitle: {
    fontSize: 16,
    fontWeight: 'bold',
  },
  helpTopicDescription: {
    color: '#666',
  },
});

export default HelpScreen;

File: Scripts/results/generated_code_results/Flutter/bak_code/Finance/Barclays/HelpScreen.dart
import 'package:flutter/material.dart';
class HelpScreen extends StatelessWidget {
@override
Widget build(BuildContext context) {
return Scaffold(
appBar: AppBar(
title: Text('Help'),
),
body: SingleChildScrollView(
child: Padding(
padding: EdgeInsets.all(16.0),
child: Column(
crossAxisAlignment: CrossAxisAlignment.start,
children: [
TextField(
decoration: InputDecoration(
hintText: 'Search or ask a question',
prefixIcon: Icon(Icons.search),
border: OutlineInputBorder(
borderRadius: BorderRadius.circular(8.0),
),
),
),
SizedBox(height: 24),
Text(
'Quick links',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
Row(
mainAxisAlignment: MainAxisAlignment.spaceAround,
children: [
_buildQuickLink(Icons.warning, 'Report fraud', () {}),
_buildQuickLink(Icons.credit_card, 'Cards', () {
Navigator.pushNamed(context, 'Cards');
}),
_buildQuickLink(Icons.description, 'Statements', () {
Navigator.pushNamed(context, 'StatementsandDocuments');
}),
],
),
SizedBox(height: 24),
Text(
'Help topics',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
ListTile(
leading: Icon(Icons.help_outline, color: Colors.blue),
title: Text('Browse all topics'),
subtitle: Text('Check how to manage Direct Debits, pay in a cheque and more.'),
trailing: Icon(Icons.arrow_forward_ios),
onTap: () {},
),
SizedBox(height: 24),
Text(
'More help',
style: TextStyle(fontSize: 18, fontWeight: FontWeight.bold),
),
SizedBox(height: 16),
_buildHelpListTile(Icons.attach_money, 'Money worries and support'),
_buildHelpListTile(Icons.accessibility_new, 'Accessible services'),
_buildHelpListTile(Icons.bar_chart, 'Our service status'),
_buildHelpListTile(Icons.play_circle_outline, 'Tour the app'),
],
),
),
),
bottomNavigationBar: BottomNavigationBar(
type: BottomNavigationBarType.fixed,
currentIndex: 3,
items: [
BottomNavigationBarItem(icon: Icon(Icons.home), label: 'Accounts'),
BottomNavigationBarItem(icon: Icon(Icons.shopping_bag), label: 'Products'),
BottomNavigationBarItem(icon: Icon(Icons.swap_horiz), label: 'Pay & Transfer'),
BottomNavigationBarItem(icon: Icon(Icons.help), label: 'Help'),
BottomNavigationBarItem(icon: Icon(Icons.more_horiz), label: 'More'),
],
onTap: (index) {
switch (index) {
case 0:
Navigator.pushNamed(context, 'Home');
break;
case 4:
Navigator.pushNamed(context, 'More');
break;
}
},
),
);
}
Widget _buildQuickLink(IconData icon, String label, VoidCallback onTap) {
return GestureDetector(
onTap: onTap,
child: Column(
children: [
Container(
width: 60,
height: 60,
decoration: BoxDecoration(
color: Colors.blue,
shape: BoxShape.circle,
),
child: Icon(icon, color: Colors.white, size: 30),
),
SizedBox(height: 8),
Text(label),
],
),
);
}
Widget _buildHelpListTile(IconData icon, String title) {
return ListTile(
leading: Icon(icon, color: Colors.blue),
title: Text(title),
trailing: Icon(Icons.arrow_forward_ios),
onTap: () {},
);
}
}